src/com/google/enterprise/adaptor/fs/FsAdaptor.java

```diff
@@ -700,7 +714,9 @@ private void validateShare(Path sharePath) throws IOException {
     // Verify that the adaptor has permission to read the Acl and share Acl.
     try {
       readShareAcls(sharePath);
-      delegate.getAclViews(sharePath);
+      if (!delegate.isDfsNamespace(sharePath) || allowFilesInDfsNamespaces) {
+        delegate.getAclViews(sharePath);
```

Comment:
I think per-startpath configuration would be a needless complication for a feature 1 customer wants. I have a sense that one of the reasons we might not be able to treat DFS Namespaces as ordinary directories is getting its filesystem ACL may fail. So this code basically preserves the current behavior.